### ISSUE TITLE
panache does not find UUID fields

### ISSUE THREAD

**Issue by stefan-langenmaier** [quarkusio/quarkus#6356]
**Describe the bug**
```
@Entity
public class Session extends PanacheEntity {

	public UUID uuid;

	public static Session findByUuid(UUID uuid){
		return find("uuid", uuid).firstResult();
	}

}
```

**Expected behavior**
the above `findByUuid` method returns a Session object when searched for a previously persisted Session

**Actual behavior**
always returns null

**To Reproduce**
1. persist session object
2. try to retrieve it by uuid
3. no object could be retrieved, it returns null

**Configuration**
```properties
# datasource configuration
quarkus.datasource.url = jdbc:mariadb://localhost:3306/mydatabase
quarkus.datasource.driver = org.mariadb.jdbc.Driver
quarkus.datasource.username = myuser
quarkus.datasource.password = mypassword

# drop and create the database at startup (use `update` to only update the schema)
quarkus.hibernate-orm.database.generation=drop-and-create
#quarkus.hibernate-orm.database.generation=update
quarkus.hibernate-orm.dialect=org.hibernate.dialect.MariaDB102Dialect
quarkus.hibernate-orm.log.sql=true
```

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: Linux tp-yp2-sl 4.19.72-gentoo #1 SMP Sun Oct 6 02:44:06 CEST 2019 x86_64 Intel(R) Core(TM) i7-4500U CPU @ 1.80GHz GenuineIntel GNU/Linux
 - Output of `java -version`: openjdk version "1.8.0_232"
OpenJDK Runtime Environment (IcedTea 3.14.0) (Gentoo icedtea-3.14.0)
OpenJDK 64-Bit Server VM (build 25.232-b09, mixed mode)
 - GraalVM version (if different from Java): not used, only jar version tested so far
 - Quarkus version or git rev: 1.1.0.Final

**Additional context**
I'm using MariaDB 10.2.

**Comment by szprajt** [comment]
This uuid is your primarykey ?

**Comment by stefan-langenmaier** [comment]
No, I use the primary key `id` provided by `PanacheEntity`. This is an additional field. I want to make it unique. I tried to reduce the example as much as possible.

**Comment by szprajt** [comment]
Yeah, just asked due PanacheEntityBase is the way to override it as a primary. 
However tried to reconstruct this problem and same happened with me too.
But it looks like for me MariaDB specific error if i need to guess due lack of UUID column types. 
Because i tried with H2 too and with that finding based on UUID properly worked so it's only occur in databases where specific UUID column type not exist.

**Comment by loicmathieu** [comment]
Did you try with `uuid.toString()` ?

**Comment by stefan-langenmaier** [comment]
Hi @loicmathieu ,

not sure if I understand your question correctly. At the moment I use `public String uuid;` for the field/column and treat it like a varchar/text column. So I can workaround the issue.

If you wanted to know if I tried `return find("uuid", uuid.toString()).firstResult();` then no, I haven't tried this, but as I saw from the type of the database column (`BINARY`) it didn't make sense to me.

**Comment by FroMage** [comment]
That's an ORM question, @Sanne @emmanuelbernard

**Comment by emmanuelbernard** [comment]
I am not sure what the problem is, so you use the `UUID` type (not string) and it works for H2 but not for MariaDB?
Can you check whether your MariaDB uuid field is not empty on creation?
In other words, is it a query problem?

**Comment by Sanne** [comment]
Did you maybe build with H2 as a dialect first, and change configuration after that?

In such cases the Dialect will still be H2; just mentioning it as we don't provide a good error/warning to such configuration issues yet.

**Comment by maweis1981** [comment]
I have the same problem now, Did this issue has been fixed yet?

**Comment by stefan-langenmaier** [comment]
I haven't tried it again. I used the workaround to store it as a string.

**Comment by ggrebert** [comment]
Confirmed. (Quarkus 2.5.2)

Works with `postgresql`, `h2`, `mssql`, `db2` 
but not for `mysql` and `mariadb`.

Project sample: https://github.com/ggrebert/quarkus-6356

Report: https://github.com/ggrebert/quarkus-6356/actions/runs/1582281708

**Comment by wicksim** [comment]
Same here with H2. If I have

` @Column private UUID externalId;`,

`repository.find("externalId", externalId).firstResult();`

does not find anything.

**Comment by oliv37** [comment]
I had the same problem, I solved it by adding `@Column(columnDefinition="uuid")` on the `UUID` field of the entity.

**Comment by yrodiere** [comment]
> Confirmed. (Quarkus 2.5.2)
> 
> Works with postgresql, h2, mssql, db2
> but not for mysql and mariadb.
> 
> Project sample: https://github.com/ggrebert/quarkus-6356
> 
> Report: https://github.com/ggrebert/quarkus-6356/actions/runs/1582281708

That definitely looks like a problem in Hibernate ORM, not in Quarkus.

Hey @beikov I know you've worked on UUIDs in Hibernate ORM recently, especially on mysql/mariadb. Does that problem (not being able to do an '=' match on UUID properties) ring a bell?

**Comment by beikov** [comment]
The UUID stuff I worked on was for Hibernate 6, but Quarkus is still on 5.6. I think this is kind of related to https://hibernate.atlassian.net/browse/HHH-15101 and https://hibernate.atlassian.net/browse/HHH-10564 in the sense that the DDL type for UUID columns must be `binary(16)`, ideally through specifying `@Column(length = 16)`.

**Comment by yrodiere** [comment]
Ok, thanks @beikov , this confirms what I suspected. 

So, this is a problem in Hibernate ORM 5, and it has been fixed in ORM 6. It will be solved in Quarkus 3 as we upgrade to ORM 6.

In the meantime, various workarounds have been suggested above (`@Column(columnDefinition="uuid")` or `@Column(length = 16)`). Feel free to have a look at https://hibernate.atlassian.net/browse/HHH-15101 (h2) or https://hibernate.atlassian.net/browse/HHH-10564 (mysql/mariadb) for more information.

I will close this as not planned, since the problem is not specific to Quarkus.
